全盘扫描异常

版本：v4.0.0

log：
eyeblue-app-1  | [INFO ]18:58:35 matter_service.go:1298 physics file not exist. delete from tank. 《消失的痕迹》 第09集 - 1080P高清完整版免费在线观看 - FreeOK - 追剧也很卷.mp4
eyeblue-app-1  | [ERROR]18:58:35 matter_dao.go:540 occur error when deleting file. remove /data/build/matter/mugu/root/Videos/《消失的痕迹》 第09集 - 1080P高清完整版免费在线观看 - FreeOK - 追剧也很卷.mp4: no such file or directory
eyeblue-app-1  | [INFO ]18:58:35 matter_service.go:1298 physics file not exist. delete from tank. code-server-4.16.1-linux-amd64.tar.gz
eyeblue-app-1  | [ERROR]18:58:36 matter_dao.go:540 occur error when deleting file. remove /data/build/matter/mugu/root/Downloads/yuhh/code-server-4.16.1-linux-amd64.tar.gz: no such file or directory
eyeblue-app-1  | [INFO ]18:58:36 matter_service.go:1298 physics file not exist. delete from tank. yuhh
eyeblue-app-1  | [ERROR]18:58:36 handler.go:7 error in async method: occur error while reading /data/build/matter/mugu/root/Downloads/yuhh open /data/build/matter/mugu/root/Downloads/yuhh: no such file or directory
eyeblue-app-1  | [INFO ]18:58:36 task_service.go:111 finish the scan task.

从磁盘直接删除一些文件和文件夹，然后进行全盘扫描，当扫描到一个被删除的文件夹（yuhh）时扫描停止，但是还有一些被删除的文件未与数据库同步

--- a/code/rest/matter_dao.go
+++ b/code/rest/matter_dao.go
@@ -518,9 +518,10 @@ func (this *MatterDao) Delete(matter *Matter) {
 		//delete from db.
 		db := core.CONTEXT.GetDB().Delete(&matter)
 		this.PanicError(db.Error)
-
-		//delete dir from disk.
-		util.DeleteEmptyDir(matter.AbsolutePath())
+		if util.PathExists(matter.AbsolutePath()) {
+			//delete dir from disk.
+			util.DeleteEmptyDir(matter.AbsolutePath())
+		}
 
 	} else {
 

--- a/code/rest/task_service.go
+++ b/code/rest/task_service.go
@@ -133,9 +133,10 @@ func (this *TaskService) doScanTask() {
 				this.logger.Info("scan spaceName = %s", space.Name)
 
 				//find admin as operator.
-				adminUser := this.userDao.FindAnAdmin()
-				this.matterService.DeleteByPhysics(request, adminUser, space)
-				this.matterService.ScanPhysics(request, adminUser, space)
+				// adminUser := this.userDao.FindAnAdmin()
+				user := this.userDao.FindByUuid(space.UserUuid)
+				this.matterService.DeleteByPhysics(request, user, space)
+				this.matterService.ScanPhysics(request, user, space)
 
 			})
 

--- a/code/rest/task_service.go
+++ b/code/rest/task_service.go
@@ -132,8 +132,7 @@ func (this *TaskService) doScanTask() {
 
 				this.logger.Info("scan spaceName = %s", space.Name)
 
-				//find admin as operator.
-				// adminUser := this.userDao.FindAnAdmin()
+				//find user by space
 				user := this.userDao.FindByUuid(space.UserUuid)
 				this.matterService.DeleteByPhysics(request, user, space)
 				this.matterService.ScanPhysics(request, user, space)
@@ -143,9 +142,6 @@ func (this *TaskService) doScanTask() {
 		})
 
 	} else if scanConfig.Scope == SCAN_SCOPE_CUSTOM {
-		//scan custom user's folder.
-		//find admin as operator.
-		adminUser := this.userDao.FindAnAdmin()
 
 		for _, spaceName := range scanConfig.SpaceNames {
 			space := this.spaceDao.FindByName(spaceName)
@@ -155,9 +151,10 @@ func (this *TaskService) doScanTask() {
 				this.logger.Info("scan custom user folder. spaceName = %s", spaceName)
 
 				core.RunWithRecovery(func() {
-
-					this.matterService.DeleteByPhysics(request, adminUser, space)
-					this.matterService.ScanPhysics(request, adminUser, space)
+					//find user by space
+					user := this.userDao.FindByUuid(space.UserUuid)
+					this.matterService.DeleteByPhysics(request, user, space)
+					this.matterService.ScanPhysics(request, user, space)
 
 				})
 

